drivers/pcnt/Kconfig

```diff
@@ -0,0 +1,19 @@
+#
+# For a description of the syntax of this configuration file,
+# see the file kconfig-language.txt in the NuttX tools repository.
+#
+
+menuconfig PCNT
+	bool "PCNT (Pulse Counter) Driver Support"
+	default n
+	---help---
+		This option selects common PCNT (Pulse Counter) options and should be
+		enabled by the platforms that implement a Pulse Counter Peripheral.
+
+if PCNT
+
+config PCNTCHAR
```

Comment:
why not use pwm capture interface?

Reply:
Pulse counter is different peripheral, it only count pulses (in a defined way). It does not uses freq or duty. For detailed information, please check [mcpwm](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/peripherals/mcpwm.html) and [pulse counter](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/peripherals/pcnt.html) on idf

Reply:
Why not add additional ioctl `CAPIOC_PULSES` to `capture.c` ? If the lower half implementation doesn't support freq or duty, then you can simply return an error code from the lower half implementation.

Reply:
If we stick to a additional driver, I think it should be named `counter` and placed in `drivers/timers` like it's done for `capture`